### PR TITLE
Fix Python support for gRPC

### PR DESCRIPTION
**Pull request by mering**
This also adds the corresponding test targets to avoid backsliding.

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (grpc) have been updated in this PR. Please review the changes.

**Comment by mering** [comment]
@fmeum could you please trigger the tests?

**Comment by mering** [comment]
@fmeum Tests are passing.

File: modules/grpc/1.62.1.bcr.2/patches/py_grpc_library.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
+diff --git a/bazel/python_rules.bzl b/bazel/python_rules.bzl
```
Comment (meteorcloudy) [inline modules/grpc/1.62.1.bcr.2/patches/py_grpc_library.patch:1]: Can we upstream this patch?

Comment (mering) [reply modules/grpc/1.62.1.bcr.2/patches/py_grpc_library.patch:1]: I sent a PR: https://github.com/grpc/grpc/pull/36843

**Comment by mering** [comment]
@meteorcloudy Anything else?

**Review by meteorcloudy — APPROVED** [review]

**Pull request merged**